### PR TITLE
FPO-382 no scenario

### PR DESCRIPTION
**Pull request by Lee365**
### Jira link
https://transformuk.atlassian.net/browse/FPO-382

FPO-323

### What?

I have added/removed/altered:

- [x] Added
- [x] Removed
- [x] Altered

### Why?

I am doing this because:

-

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: src/controllers/identifyController.ts [hunk 1]
```diff
@@ -36,12 +36,13 @@ export class IdentifyController {
         answer
       }
     })
-    console.log(newIdentification.id)
+    console.log('newIdentification:' + newIdentification.id)
```
Comment (mkayad) [inline src/controllers/identifyController.ts:39]: I think you can remove this log

File: src/controllers/reasonForNoController.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { type Request, type Response } from 'express'
+import Identification from '../models/identification'
+
+export class ReasonForNoController {
+  public show (req: Request, res: Response): void {
+    if (req.body.reasoneForNoCode === 'vague') {
```
Comment (willfish) [inline src/controllers/reasonForNoController.ts:6]:
```suggestion
    if (req.body.reasonForNoCode === 'vague') {
```

File: src/controllers/reasonForNoController.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { type Request, type Response } from 'express'
+import Identification from '../models/identification'
+
+export class ReasonForNoController {
```
Comment (mkayad) [inline src/controllers/reasonForNoController.ts:4]: To be consistent, I will rename this as ImproveController as we have IdentifyController

File: src/controllers/reasonForNoController.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      const session = req.session ?? {}
+      res.render('correctionForNoScenario', { session })
+    }
+  }
+
+  public async create (req: Request, res: Response): Promise<void> {
+    const session = req.session ?? {}
+    const newIdentificationId = session.newIdentificationId
+
+    await Identification.upsert({
```
Comment (mkayad) [inline src/controllers/reasonForNoController.ts:20]: Do we need to update the whole record? can we just update the answer with the new commodity code?

File: views/correctionForNoScenario.njk [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{% extends "layout.njk" %}
```
Comment (mkayad) [inline views/correctionForNoScenario.njk:1]: would be good to rename this file to something more meaningful

File: views/reasonForNoScenario.njk [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{% extends "layout.njk" %}
```
Comment (mkayad) [inline views/reasonForNoScenario.njk:1]: would be good to rename this file to something more meaningful

File: src/controllers/reasonForNoController.ts (hunk 4 above)
Comment (willfish) [inline src/controllers/reasonForNoController.ts:20]: Don't we just want to do an update of the already existing row:

```node
await Identification.update(
  {
    answer: {
      no: {
        reason: req.body.reasoneForNoCode,
        explanation: req.body.explanation
      }
    }
  },
  {
    where: {
      id: newIdentificationId
    }
  }
);
```

File: src/routes/index.ts [hunk 7]
```diff
@@ -19,4 +21,6 @@ export default router.get('/', (_req: Request, res: Response, _next: NextFunctio
 /* eslint-disable  @typescript-eslint/no-floating-promises */
 router.get('/identify', (req, res) => { identifyController.show(req, res) })
 router.post('/save', (req, res) => { identifyController.create(req, res) })
+router.post('/saveReasonForNo', (req, res) => { reasonForNoController.show(req, res) })
```
Comment (willfish) [inline src/routes/index.ts:24]: Its actually simpler/avoids collisions if we just use restful endpoints.

Comment (willfish) [reply src/routes/index.ts:24]:
```
GET     /identifications            # Fetch all identifications
POST    /identifications            # Create a new identification
GET     /identifications/:id        # Fetch a specific identification by ID
PUT     /identifications/:id        # Update an entire identification by ID
PATCH   /identifications/:id        # Partially update an identification by ID
DELETE  /identifications/:id        # Delete an identification by ID
```

Comment (willfish) [reply src/routes/index.ts:24]: So what you're doing here is PATCHING an existing row

File: src/controllers/identifyController.ts (hunk 1 above)
Comment (Lee365) [reply src/controllers/identifyController.ts:39]: removed

File: src/controllers/reasonForNoController.ts (hunk 3 above)
Comment (Lee365) [reply src/controllers/reasonForNoController.ts:4]: ImproveController

File: src/controllers/reasonForNoController.ts (hunk 2 above)
Comment (Lee365) [reply src/controllers/reasonForNoController.ts:6]: Fixed

File: src/controllers/reasonForNoController.ts (hunk 4 above)
Comment (Lee365) [reply src/controllers/reasonForNoController.ts:20]: Only answer updated

Comment (Lee365) [reply src/controllers/reasonForNoController.ts:20]: Only answer updated

File: views/correctionForNoScenario.njk (hunk 5 above)
Comment (Lee365) [reply views/correctionForNoScenario.njk:1]: renamed

File: views/reasonForNoScenario.njk (hunk 6 above)
Comment (Lee365) [reply views/reasonForNoScenario.njk:1]: renamed

**Pull request merged**